.changeset/great-kids-visit.md

```diff
@@ -0,0 +1,5 @@
+---
+'@apollo/server': patch
+---
+
+Add an `__identity` property to `HeaderMap` class to disallow standard `Map`s (in TypeScript)
```

Comment:
Maybe this should be more like "Ensure via typechecking that fields declared to take `HeaderMap` (notably, the `httpGraphQLRequest.headers` option to `ApolloServer.executeHTTPGraphQLRequest` and the `http.headers` option to `ApolloServer.executeOperation`) actually use a `HeaderMap` rather than allowing other `Map<string, string>`s. This may be a breaking change for framework integrations, but it is easily fixed by switching from `new Map<string, string>()` to `new HeaderMap()` and those integrations previously had bugs" or something.